libs/ui-watt/src/lib/components/input/timepicker/watt-timepicker.component.ts

```diff
@@ -121,23 +226,45 @@ export class WattTimepickerComponent extends WattPickerBase {
    */
   protected initRangeInput() {
     // Setup and subscribe for input changes
-    this.rangeInputService.init({
-      startInput: this.maskInput(
-        this.startInput.nativeElement,
-        (this.initialValue as WattRange | null)?.start
-      ),
-      endInput: this.maskInput(
-        this.endInput.nativeElement,
-        (this.initialValue as WattRange | null)?.end
-      ),
+    const startInput = this.maskInput(
+      this.startInput.nativeElement,
+      (this.initialValue as WattRange | null)?.start
+    );
+
+    const endInput = this.maskInput(
+      this.endInput.nativeElement,
+      (this.initialValue as WattRange | null)?.end
+    );
+
+    this.rangeInputService.init({ startInput, endInput });
+
+    // Silence the compiler since `onInputChanges$` is always assigned in `init`
+    const { onInputChanges$ = EMPTY } = this.rangeInputService;
+    const timeRange$ = onInputChanges$.pipe(takeUntil(this.destroy$));
+
+    // Synchronize the slider value with the input fields. Calling `update`
+    // here automatically triggers an emit on the `timeRange$` observable.
+    this.sliderChange$.pipe(takeUntil(this.destroy$)).subscribe((value) => {
+      startInput.maskedInput.update(minutesToTime(value.min));
+      endInput.maskedInput.update(minutesToTime(value.max));
     });
 
-    this.rangeInputService.onInputChanges$
-      ?.pipe(takeUntil(this.destroy$))
-      .subscribe(([start, end]) => {
-        this.markParentControlAsTouched();
-        this.changeParentValue({ start, end });
-      });
+    // Whenever the start input value changes, set (or remove)
+    // the minimum allowed value for the end input value.
+    timeRange$
+      .pipe(map(getTruthy(0)), distinctUntilChanged())
+      .subscribe((start) => endInput.maskedInput.setOptions({ min: start }));
+
+    // Whenever the end input value changes, set (or remove)
+    // the maximum allowed value for the start input value.
+    timeRange$
+      .pipe(map(getTruthy(1)), distinctUntilChanged())
```

Comment:
suggestion: wouldn't this give the same result?

```suggestion
      .pipe(map(([, end]) => end || undefined), distinctUntilChanged())
```

Reply:
See above.